src/core/space_interaction_API.jl

```diff
@@ -371,3 +372,38 @@ function random_nearby_agent(a, model, r = 1; kwargs...)
     isnothing(id) && return
     return model[id]
 end
+
+"""
+    random_nearby_position(position, model::ABM, r=1; kwargs...) → position
+
+Return a random position near the given `position`. Return `nothing` if the space doesn't allow for nearby positions.
+
+The value of the argument `r` and possible keywords operate identically to [`nearby_positions`](@ref).
+"""
+function random_nearby_position(pos, model, r=1; kwargs...)
+    # Uses the same Reservoir Sampling algorithm than nearby_ids
+    iter = nearby_positions(pos, model, r; kwargs...)
+
+    res = iterate(iter)
+    isnothing(res) && return    # `iterate` returns `nothing` when it ends
```

Comment:
```suggestion
    isnothing(res) && return nothing    # `iterate` returns `nothing` when it ends
```
Better explicit returns than implicit ones